sql/analyzer/resolve_natural_joins.go

```diff
@@ -135,9 +149,14 @@ func resolveNaturalJoins(ctx *sql.Context, a *Analyzer, n sql.Node) (sql.Node, e
 		return node, err
 	}
 
+	var transformedSeen bool
 	return node.TransformUp(func(node sql.Node) (sql.Node, error) {
-		above, colsToUnresolve := isOverTranformedNode(node, transformed)
-		if !above {
+		if ok, _ := isTransformedNode(node, transformed); ok {
```

Comment:
here I think we can simplify by:
```
transformedSeen, _ = isTransformedNode(node, transformed)
return node, nil
```

Reply:
I can't do that. `transformedSeen` is a variable outside the scope of the function executed for each node by the `node.TransformUp` method.

If I always assign the `transformedSeen` variable, it will be false the next time a non transformed node is evaluated although a previous node is a transformed node.

I'm not sure if I explained myself well or if I misunderstood you @kuba--

Reply:
Ok, I know what you mean - nvm.